internal/ansible/controller/reconcile.go

```diff
@@ -100,22 +101,21 @@ func (r *AnsibleOperatorReconciler) Reconcile(ctx context.Context, request recon
 
 	deleted := u.GetDeletionTimestamp() != nil
 	finalizer, finalizerExists := r.Runner.GetFinalizer()
-	pendingFinalizers := u.GetFinalizers()
-	// If the resource is being deleted we don't want to add the finalizer again
-	if finalizerExists && !deleted && !contains(pendingFinalizers, finalizer) {
-		logger.V(1).Info("Adding finalizer to resource", "Finalizer", finalizer)
-		finalizers := append(pendingFinalizers, finalizer)
-		u.SetFinalizers(finalizers)
-		err := r.Client.Update(ctx, u)
-		if err != nil {
-			logger.Error(err, "Unable to update cr with finalizer")
-			return reconcileResult, err
+	if !controllerutil.ContainsFinalizer(u, finalizer) {
+		if deleted {
+			// If the resource is being deleted we don't want to add the finalizer again
```

Comment:
Shouldn't we add the finalizer here? If the object has deletion timestamp on it and doesn't have finalizer on it, we add it, right? Unless I misunderstood this. :)

Reply:
All CRs get a finalizer as soon as they're reconciled for the first time. That finalizer only gets removed once a deletion event occurs.

Reply:
Ah I see. That makes sense.